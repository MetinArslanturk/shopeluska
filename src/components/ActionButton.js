import React from 'react';
import { Button } from 'antd';

const ActionButton = ({ type, text, onClickActionFunction, itemToArguments }) => {
    const handleOnclick = () => {
        onClickActionFunction(itemToArguments);
    };
console.log('rendered');
    return (
        <>
            <Button type={type} className="action-button" onClick={handleOnclick}>{text}</Button>
        </>
    );

}

export default React.memo(ActionButton);