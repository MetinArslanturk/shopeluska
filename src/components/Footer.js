import React from 'react';

const Footer = (props) => {
    return (
        <>
            <div>Copyright to <a rel="noopener noreferrer" target="_blank" href="http://www.metinarslanturk.com">Metin Arslanturk</a> You can check <a rel="noopener noreferrer" target="_blank" href="https://github.com/MetinArslanturk/shopeluska">GitHub of This Project</a></div>
        </>
    );
}



export default Footer;