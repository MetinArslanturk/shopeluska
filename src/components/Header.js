import React, { useState } from 'react';
import { connect } from 'react-redux';
import { history, baseHref } from '../config/config';
import { Drawer, Row, Col, Button, Input } from 'antd';
import { Link } from 'react-router-dom';
import { BarsOutlined } from '@ant-design/icons';
import { startLogout } from '../actions/auth';
import {getUserName, isAuthenticated} from '../selectors/auth';
const { Search } = Input;

const clickLogin = () => {
    history.push(baseHref + 'login');
}

const clickMyAccount = () => {
    history.push(baseHref + 'my-account/my-profile');
}

export const Header = ({ isAuthenticated, username, startLogout}) => {
    const [visibleNav, setVisibleNav] = useState(false);

    return (
        <>
            <div className="header-wrapper">

                <Row>
                    <Col span={4}>
                        <div className="brand-logo">
                            <Link to={baseHref}><img alt="logo" className="logo-image" src="logo.png" /></Link>
                        </div>
                    </Col>
                    <Col xs={4} sm={4} md={7} lg={8} xl={12}>
                        <div className="search-box">
                            <Search placeholder="input search text" onSearch={value => console.log(value)} enterButton />
                        </div>
                    </Col>
                    <Col xs={4} sm={6} md={6} lg={6} xl={4}>
                        <div className="user-nav">
                            {!isAuthenticated && (
                                <Button type="primary" className="login-button" onClick={clickLogin}>Login</Button>
                            )}
                            {isAuthenticated && (
                                <>
                                    <Button className="my-account-button" type="primary" onClick={clickMyAccount}>My Account</Button>
                                    <Button className="logout-button" onClick={startLogout}>Logout</Button>
                                </>
                            )}
                        </div>
                        <div className="head-menu-toggle-button">
                            <BarsOutlined onClick={e => setVisibleNav(true)} style={{ fontSize: '32px', marginTop: '18px' }} />
                        </div>
                    </Col>


                </Row>
            </div>
            <Drawer
                title="Shopeluska Navigation"
                placement="right"
                closable={true}
                onClose={() => {
                    setVisibleNav(false);
                }}
                visible={visibleNav}
            >
                <p>Some contents...</p>
                <p>Some contents...</p>
                <p>Some contents...</p>
            </Drawer>
        </>
    )
}
const mapStateToProps = (state) => ({
    isAuthenticated: isAuthenticated(state.auth.user),
    username: getUserName(state.auth.user)
});

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(mapStateToProps, mapDispatchToProps)(Header);