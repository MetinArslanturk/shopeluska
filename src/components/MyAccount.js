import React, { useState } from 'react';
import { connect } from 'react-redux'
import MyOrdersPage from './my-account/MyOrdersPage';
import MyProfilePage from './my-account/MyProfilePage';
import AdminPage from './my-account/AdminPage';
import { isAdmin } from '../selectors/auth';
import { Tabs } from 'antd';

const { TabPane } = Tabs;

export const MyAccount = ({ isAdmin }) => {
    const [selectedPage, setPage] = useState('my-profile');

    return (
        <>
            <Tabs type="card" defaultActiveKey={selectedPage} onChange={setPage}>
                <TabPane tab="My Profile" key="my-profile"></TabPane>
                <TabPane tab="My Orders" key="my-orders"></TabPane>
                {isAdmin && <TabPane tab="Admin" key="admin"></TabPane>}
            </Tabs>

            {selectedPage === 'my-profile' ? (
                <>
                    <MyProfilePage />
                </>
            ) : selectedPage === 'my-orders' ? (
                <>
                    <MyOrdersPage />
                </>
            ) : (selectedPage === 'admin' && isAdmin) ? (
                <>
                    <AdminPage />
                </>
            ) : (<><div>Upss.. Something went wrong...</div></>)}
        </>
    )
}

const mapStateToProps = (state) => ({
    isAdmin: isAdmin(state.auth.user)
})

export default connect(mapStateToProps)(MyAccount);