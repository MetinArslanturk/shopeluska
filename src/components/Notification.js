import { notification } from 'antd';

const openNotification = (title = '', description = '', duration = 0, type = 'info') => {
    const args = {
      message: title,
      description,
      duration
    };
    notification[type](args);
  };

  export const showInfoNotification = (title, desc, duration) => {
    openNotification(title, desc, duration, 'info');
  };

  export const showErrorNotification = (title, desc, duration) => {
    openNotification(title, desc, duration, 'error');
  };

  export const showSuccessNotification = (title, desc, duration) => {
    openNotification(title, desc, duration, 'success');
  };

  export const showWarningNotification = (title, desc, duration) => {
    openNotification(title, desc, duration, 'warning');
  };

  export const destroyNotifications = () => {
    notification.destroy();
  }

  
