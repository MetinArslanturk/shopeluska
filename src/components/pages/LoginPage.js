import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { baseHref } from '../../config/config';
import { useMountEffect } from '../../helpers/mounteffect';
import { hideSidebar, showSidebar } from '../../actions/layout';
import { startLogin } from '../../actions/auth';
import { isAuthenticated } from '../../selectors/auth';
import { Form, Input, Button } from 'antd';

const layout = {
    labelCol: { span: 8 },
    wrapperCol: { span: 16 },
};
const tailLayout = {
    wrapperCol: { offset: 8, span: 16 },
};


export const LoginPage = ({ hideSidebar, showSidebar, startLogin, isAuthenticated, redirectUrl }) => {
    useMountEffect(() => {
        hideSidebar();
        return () => {
            // Give back the sidebar
            showSidebar();
        }
    })

    const onFinish = values => {
        // dummy start login
        startLogin(values);
    };


    return (
        <>
            {isAuthenticated ? (
                <Redirect to={redirectUrl ? redirectUrl : baseHref} />
            ) : (
                    <div className="login-wrapper">
                        <div className="login-form">
                            <Form
                                {...layout}
                                name="basic"
                                initialValues={{ remember: true }}
                                onFinish={onFinish}
                                id="loginform"
                            >
                                <Form.Item
                                    label="Email"
                                    name="email"
                                    rules={[{ required: true, message: 'Please input your email!' }]}
                                >
                                    <Input />
                                </Form.Item>

                                <Form.Item
                                    label="Password"
                                    name="password"
                                    rules={[{ required: true, message: 'Please input your password!' }]}
                                >
                                    <Input.Password />
                                </Form.Item>


                                <Form.Item {...tailLayout}>
                                    <Button type="primary" htmlType="submit">
                                        Submit
                        </Button>
                                </Form.Item>
                            </Form>
                        </div>
                    </div>
                )}

        </>
    );
}

const mapStateToProps = (state) => ({
    isAuthenticated: isAuthenticated(state.auth.user),
    redirectUrl: state.auth.redirectUrl
});

const mapDispatchToProps = (dispatch) => ({
    hideSidebar: () => dispatch(hideSidebar()),
    showSidebar: () => dispatch(showSidebar()),
    startLogin: (values) => dispatch(startLogin(values))
})
export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);