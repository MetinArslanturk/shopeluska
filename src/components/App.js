import React from 'react';
import { connect } from 'react-redux'
import Header from './layouts/Header';
import Sidebar from './layouts/Sidebar';
import Footer from './layouts/Footer';
import { checkLogin } from '../actions/auth';
import { startSetProducts } from '../actions/products';
import { Layout } from 'antd';
import AppRouter from '../routers/AppRouter';


const { Header: LayoutHeader, Footer: LayoutFooter, Content } = Layout;


class App extends React.Component {
    componentDidMount() {
        this.props.dispatch(startSetProducts());
        this.props.dispatch(checkLogin());
    }

    render() {
        return (
            <>
                <Layout>
                    <LayoutHeader>
                        <Header />
                    </LayoutHeader>
                    <Layout>
                        <Sidebar />
                        <Content>
                            <AppRouter />
                        </Content>
                    </Layout>
                    <LayoutFooter>
                        <Footer />
                    </LayoutFooter>
                </Layout>
            </>
        );
    }
}


export default connect(undefined)(App);
