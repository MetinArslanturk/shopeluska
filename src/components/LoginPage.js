import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { baseHref } from '../config/config';
import { useMountEffect } from '../helpers/mounteffect';
import { hideSidebar, showSidebar } from '../actions/layout';
import { startLogin } from '../actions/auth';
import { showErrorNotification } from './Notification';
import { isAuthenticated } from '../selectors/auth';
import { Form, Input, Button } from 'antd';

const layout = {
    labelCol: { span: 8 },
    wrapperCol: { span: 16 },
};
const tailLayout = {
    wrapperCol: { offset: 8, span: 16 },
};


export const LoginPage = ({ hideSidebar, showSidebar, startLogin, isAuthenticated, redirectPath }) => {
    useMountEffect(() => {
        hideSidebar();
        return () => {
            // Give back the sidebar
            showSidebar();
        }
    })

    const onFinish = values => {
        // dummy start login
    //    showErrorNotification('Error', 'Given username or password is incorrect!', 0);
        startLogin();
    };


    return (
        <>
            {isAuthenticated ? (
                <Redirect to={redirectPath ? redirectPath : baseHref} />
            ) : (
                    <div className="login-wrapper">
                        <div className="login-form">
                            <Form
                                {...layout}
                                name="basic"
                                initialValues={{ remember: true }}
                                onFinish={onFinish}
                                id="loginform"
                            >
                                <Form.Item
                                    label="Username"
                                    name="username"
                                    rules={[{ required: true, message: 'Please input your username!' }]}
                                >
                                    <Input />
                                </Form.Item>

                                <Form.Item
                                    label="Password"
                                    name="password"
                                    rules={[{ required: true, message: 'Please input your password!' }]}
                                >
                                    <Input.Password />
                                </Form.Item>


                                <Form.Item {...tailLayout}>
                                    <Button type="primary" htmlType="submit">
                                        Submit
                        </Button>
                                </Form.Item>
                            </Form>
                        </div>
                    </div>
                )}

        </>
    );
}

const mapStateToProps = (state) => ({
    isAuthenticated: isAuthenticated(state.auth.user)
});

const mapDispatchToProps = (dispatch) => ({
    hideSidebar: () => dispatch(hideSidebar()),
    showSidebar: () => dispatch(showSidebar()),
    startLogin: () => dispatch(startLogin())
})
export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);