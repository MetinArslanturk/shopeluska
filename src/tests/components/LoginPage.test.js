import React from 'react';
import {LoginPage} from '../../components/LoginPage';
import { shallow } from 'enzyme';



test('should render login page when non-authenticated', () => {
    const startLogin = jest.fn();

    const wrapper = shallow(<LoginPage isAuthenticated={false} startLogin={startLogin} />)
    expect(wrapper).toMatchSnapshot();

    
    wrapper.find('#loginform').prop('onFinish')({username: 'aa', password:'123'});
    expect(startLogin).toHaveBeenCalled();
});


test('should redirect when authenticated', () => {
    const prop = {
        isAuthenticated: true
    }
    const wrapper = shallow(<LoginPage {...prop} />)
    expect(wrapper).toMatchSnapshot();
});