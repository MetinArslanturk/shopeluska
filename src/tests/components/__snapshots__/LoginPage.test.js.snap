// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should redirect when authenticated 1`] = `
<Fragment>
  <Redirect
    to="/shopeluska/"
  />
</Fragment>
`;

exports[`should render login page when non-authenticated 1`] = `
<Fragment>
  <div
    className="login-wrapper"
  >
    <div
      className="login-form"
    >
      <ForwardRef(InternalForm)
        id="loginform"
        initialValues={
          Object {
            "remember": true,
          }
        }
        labelCol={
          Object {
            "span": 8,
          }
        }
        name="basic"
        onFinish={[Function]}
        wrapperCol={
          Object {
            "span": 16,
          }
        }
      >
        <FormItem
          label="Email"
          name="email"
          rules={
            Array [
              Object {
                "message": "Please input your email!",
                "required": true,
              },
            ]
          }
        >
          <Input
            type="text"
          />
        </FormItem>
        <FormItem
          label="Password"
          name="password"
          rules={
            Array [
              Object {
                "message": "Please input your password!",
                "required": true,
              },
            ]
          }
        >
          <Password
            action="click"
            visibilityToggle={true}
          />
        </FormItem>
        <FormItem
          wrapperCol={
            Object {
              "offset": 8,
              "span": 16,
            }
          }
        >
          <Button
            block={false}
            ghost={false}
            htmlType="submit"
            loading={false}
            type="primary"
          >
            Submit
          </Button>
        </FormItem>
      </ForwardRef(InternalForm)>
    </div>
  </div>
</Fragment>
`;
